software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -395,12 +390,12 @@ public void joinCluster(String nodeName) {
             if (!hasJoinedCluster()) {
                 ScriptHelper joinClusterScript = newScript("joinCluster")
                         .body.append(sudo(format("%s cluster join %s", getRiakAdminCmd(), nodeName)))
+                        .body.append(sudo(format("%s cluster plan", getRiakAdminCmd())))
+                        .body.append(sudo(format("%s cluster commit", getRiakAdminCmd())))
```

Comment:
agree, it feels like our default behaviour should be to commit.

we could (in time) add an optional parameter "commitChange" which defaults to true, that way if someone wanted to they could batch/preview changes.

i'd say add a TODO with this (and reference the commit which includes the commitCluster effector, in case we want to revive that we can copy-paste from here.)